qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -173,12 +172,18 @@ def retrieve_jobs(self, provider: AccountProvider, refresh: bool = False) -> Non
         jobs_dict = {}
         for job in jobs:
             # Verify the job is proper.
-            matched = pattern.match(job.name()) if job.name() else None
-            if not matched or matched.group(1) != self._name or \
-                    job.backend().name() != self._backend.name():
+            matched = JOB_SET_NAME_RE.match(job.name()) if job.name() else None
+            if not matched or job.backend().name() != self._backend.name():
                 raise IBMQJobManagerInvalidStateError(
                     'Job {} is tagged for the job set {} but does not appear '
                     'to belong to the set.'.format(job.job_id(), self.job_set_id()))
+            if matched.group(1) != self._name:
+                logger.warning('The job %s, belonging to job set %s, does not appear to have '
+                               'a proper job name. The job name should include the job set name '
+                               '"%s", but instead includes the job set name "%s". Use the '
+                               'update_name() method to update the name of the jobs in the job set',
+                               job.job_id(), self.job_set_id(), self.name(), matched.group(1))
```

Comment:
Yeah, it's a bit difficult to know whether a job belongs to a job set since certain attributes could be updated at any time. Would it be more reasonable to check whether the job contains the job set tag, in order to determine whether the job belongs to the job set?

Reply:
The job set tag is already used to determine whether a job belongs to a set. Checking for job name and backend name is just extra precaution. One possible strategy is to collect all the job names and see if they all have the same prefix. If so, deem that as the name of the set. If not, set the job set name to None and issue a warning.

Reply:
As mentioned offline, I kept the original behavior of enforcing that all jobs have a proper name and raising an error if a job does not.